File: src/main/java/com/timyr_tm/rust_bound/RustBound.java
package com.timyr_tm.rust_bound;

import net.neoforged.bus.api.IEventBus;
import net.neoforged.fml.ModContainer;
import net.neoforged.fml.ModList;
import net.neoforged.fml.common.Mod;

import static com.timyr_tm.rust_bound.world.item.Items.ITEMS;
import static com.timyr_tm.rust_bound.world.item.CreativeModeTabs.CREATIVE_MODE_TABS;
import static com.timyr_tm.rust_bound.world.block.Blocks.BLOCKS;
import static com.timyr_tm.rust_bound.world.block.entity.BlockEntityTypes.BLOCK_ENTITY_TYPES;
import static com.timyr_tm.rust_bound.core.component.DataComponents.DATA_COMPONENTS;

@Mod(RustBound.MOD_ID)
public class RustBound {
    public static final String MOD_ID = "rust_bound";

    public RustBound(IEventBus bus, ModContainer container) {
        ITEMS.register(bus);
        CREATIVE_MODE_TABS.register(bus);
        BLOCKS.register(bus);
        BLOCK_ENTITY_TYPES.register(bus);
        DATA_COMPONENTS.register(bus);
    }

    public static ModContainer getModContainer() {
        return ModList.get().getModContainerById(MOD_ID).orElseThrow();
    }
}

File: src/main/kotlin/com/timyr_tm/rust_bound/RustBound.kt
package com.timyr_tm.rust_bound;

import net.neoforged.bus.api.IEventBus;
import net.neoforged.fml.ModContainer;
import net.neoforged.fml.common.Mod;

import com.timyr_tm.rust_bound.world.item.Items.ITEMS;
import com.timyr_tm.rust_bound.world.item.CreativeModeTabs.CREATIVE_MODE_TABS;
import com.timyr_tm.rust_bound.world.block.Blocks.BLOCKS;
import com.timyr_tm.rust_bound.world.block.entity.BlockEntityTypes.BLOCK_ENTITY_TYPES;
import com.timyr_tm.rust_bound.core.component.DataComponents.DATA_COMPONENTS;
import com.timyr_tm.rust_bound.world.electricity.WireTypes.WIRE_TYPES;

@Mod(RustBound.MOD_ID)
class RustBound {
    companion object {
        const val MOD_ID: String = "rust_bound"
    }

    constructor(bus: IEventBus, container: ModContainer) {
        ITEMS.register(bus);
        CREATIVE_MODE_TABS.register(bus);
        BLOCKS.register(bus);
        BLOCK_ENTITY_TYPES.register(bus);
        DATA_COMPONENTS.register(bus);
        WIRE_TYPES.register(bus);
    }
}
